src/averageAges.js

```diff
@@ -55,10 +78,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter((person, _, arr) => {
+    const hasMother = arr.find(mother => mother.name === person.mother);
+
+    return hasMother && (onlyWithSon ? person.sex === 'm' : true);
+  });
+
+  const childrensTotalAge = children.reduce((acc, child,) => {
```

Comment:
The reduce function's accumulator is not returned outside of the conditional statement, which can lead to `undefined` being returned unintentionally. Ensure that `acc` is returned at the end of the reduce callback function.